### ISSUE TITLE
can't save keras model built by functional api

### ISSUE THREAD

**Issue by littlehome-eugene** [tensorflow/tensorflow#27503]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): n/a
- TensorFlow version (use command below): 2.0.0-alpha0
- Python version: 3.6
- Bazel version (if compiling from source): na
- GCC/Compiler version (if compiling from source): na
- CUDA/cuDNN version: na
- GPU model and memory: na

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**

> lib/python3.6/site-packages/tensorflow/python/util/serialization.py", line 69, in get_json_type
>     raise TypeError('Not JSON Serializable:', obj)
> TypeError: ('Not JSON Serializable:', b"\n\x04Mean\x12\x04Mean\x1a'embedding_1/embedding_lookup/Identity_2\x1a\x16Mean/reduction_indices*\x07\n\x01T\x12\x020\x01*\n\n\x04Tidx\x12\x020\x03*\x0f\n\tkeep_dims\x12\x02(\x01")

**Describe the expected behavior**
model.save(path) works with a keras model

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
```

import copy
 import tensorflow as tf
 import pandas as pd
 from tensorflow.keras.layers import Input, Embedding, concatenate, Dense, Flatten
 from tensorflow import feature_column
 from tensorflow.python.keras.engine import training_utils


 def df_to_dataset(dataframe, shuffle=True, batch_size=32):
   dataframe = dataframe.copy()
   labels = dataframe.pop('target')
   d = dict(dataframe)

   ds = tf.data.Dataset.from_tensor_slices((d, labels))
   if shuffle:
     ds = ds.shuffle(buffer_size=len(dataframe))
   ds = ds.batch(batch_size)
   return ds


 def get_model():
     first_input = Input(shape = (1,), name='first_input')
     second_input = Input(shape = (1,), name='second_input' )
     embedding_layer = Embedding(input_dim=10, output_dim=3, input_length=1)

     first_input_encoded = embedding_layer(first_input)
     first_input_encoded = tf.keras.layers.Reshape((3,))(first_input_encoded)


     selected = embedding_layer(second_input)
     item_average = tf.reduce_mean(selected, axis=1, keepdims=True)
     second_input_encoded = tf.keras.layers.Reshape((3,))(item_average)


     o = concatenate([first_input_encoded, second_input_encoded])
     o = Dense(1)(o)

     inputs = [first_input, second_input]
     model = tf.keras.models.Model(inputs=inputs, outputs=o)
     return model


 df = pd.DataFrame(
     [[3,[4,2,3],5, 'boy', 0],
     [2,[6,1,2],7, 'girl', 1]], columns=['first_input', 'second_input', 'child_month_young', 'child_gender_young', 'target'])

 train_ds = df_to_dataset(df)
 val_ds = df_to_dataset(df)

 model = get_model()

 model.compile(optimizer='adam',
               loss='binary_crossentropy',
               metrics=['accuracy'])


 model.fit(
     train_ds,
     validation_data=val_ds,
     epochs=1
 )

 print(model.summary())

 # tf.keras.models.save_model(model, 'test.h5')
 model.save('test.h5')
```

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by ShaneTian** [comment]
**I have same question as you, could you tell me how did you solve it?**
---
```
TypeError: ('Not JSON Serializable:', b'\n.model_1/ExpandDims/model/ExpandDims/ExpandDims\x12\nExpandDims\x1a%embedding/embedding_lookup/Identity_2\x1a\x0eExpandDims/dim*\n\n\x04Tdim\x12\x020\x03*\x07\n\x01T\x12\x020\x01')
```

**Comment by ricosr** [comment]
> ## **I have same question as you, could you tell me how did you solve it?**
> ```
> TypeError: ('Not JSON Serializable:', b'\n.model_1/ExpandDims/model/ExpandDims/ExpandDims\x12\nExpandDims\x1a%embedding/embedding_lookup/Identity_2\x1a\x0eExpandDims/dim*\n\n\x04Tdim\x12\x020\x03*\x07\n\x01T\x12\x020\x01')
> ```

I have the same trouble as you, have you solved it?

**Comment by ricosr** [comment]
> ## **I have same question as you, could you tell me how did you solve it?**
> ```
> TypeError: ('Not JSON Serializable:', b'\n.model_1/ExpandDims/model/ExpandDims/ExpandDims\x12\nExpandDims\x1a%embedding/embedding_lookup/Identity_2\x1a\x0eExpandDims/dim*\n\n\x04Tdim\x12\x020\x03*\x07\n\x01T\x12\x020\x01')
> ```

@ShaneTian I have the same trouble as you, have you solved it?

**Comment by ShaneTian** [comment]
> > ## **I have same question as you, could you tell me how did you solve it?**
> > ```
> > TypeError: ('Not JSON Serializable:', b'\n.model_1/ExpandDims/model/ExpandDims/ExpandDims\x12\nExpandDims\x1a%embedding/embedding_lookup/Identity_2\x1a\x0eExpandDims/dim*\n\n\x04Tdim\x12\x020\x03*\x07\n\x01T\x12\x020\x01')
> > ```
> 
> @ShaneTian I have the same trouble as you, have you solved it?

Don't use 'expanddims' in Keras model, it's not compatible @ricosr

**Comment by ricosr** [comment]
> > > ## **I have same question as you, could you tell me how did you solve it?**
> > > ```
> > > TypeError: ('Not JSON Serializable:', b'\n.model_1/ExpandDims/model/ExpandDims/ExpandDims\x12\nExpandDims\x1a%embedding/embedding_lookup/Identity_2\x1a\x0eExpandDims/dim*\n\n\x04Tdim\x12\x020\x03*\x07\n\x01T\x12\x020\x01')
> > > ```
> > 
> > 
> > @ShaneTian I have the same trouble as you, have you solved it?
> 
> Don't use 'expanddims' in Keras model, it's not compatible @ricosr

@ShaneTian Thank you so much! Actually my problem is a little different from yours, I used tf.reshape in my model, but both of them are similar compatible troubles. Now I change tf.reshape by tf.keras.layers.Reshape, which can work well.

**Comment by williamrosengren** [comment]
I have also had this problem drive me mad when trying to use json to load my Keras model. I used a different approach that is slightly more primitive. Instead of using the save_model method I stored each set of weights using the model.layers and layer.get_weights() commands after training the model, storing them using cPickle. Then I reinitialised each layer by assigning the weights when adding a new layer to a completely new model in a different script:
model = Sequential()
model.add(Dense(M, weights=[Ww, Wb], input_shape=[M])). 
where Wb is the bias vector and Ww is the weight matrix. 
 
Hope this helps!
